tests/plugins/os/unix/linux/android/test__os.py

```diff
@@ -8,3 +12,49 @@ def test_android_os(target_android: Target) -> None:
     assert target_android.os == "android"
     assert target_android.version == "Android 4.4.2 (2013-10-31)"
     assert target_android.hostname == "TMG28071935"
+
+
+normal_prop = absolute_path("_data/plugins/os/unix/linux/android/build.prop")
+another_prop = absolute_path("_data/plugins/os/unix/linux/android/another.prop")
+
+
+@pytest.mark.parametrize(
+    "build_prop_locations",
+    [
+        ([("/build.prop", normal_prop)]),
+        ([("/system/build.prop", normal_prop)]),
+        ([("/build.prop", normal_prop), ("/foo/build.prop", another_prop)]),
+    ],
+)
+def test_android_os_detect_props(target_bare: Target, build_prop_locations: list[tuple[str, str]]) -> None:
+    """test if we detect different build.prop locations correctly."""
+
+    fs = VirtualFilesystem()
+    fs.makedirs("/data")
+    fs.makedirs("/system")
+    fs.makedirs("/vendor")
+    fs.makedirs("/product")
+
+    for prop, prop_file in build_prop_locations:
+        fs.map_file(prop, prop_file)
```

Comment:
```suggestion
@pytest.mark.parametrize(
    "build_prop_locations",
    [
        ([("/build.prop", "build.prop")]),
        ([("/system/build.prop", "build.prop")]),
        ([("/build.prop", "build.prop"), ("/foo/build.prop", "another.prop")]),
    ],
)
def test_android_os_detect_props(target_bare: Target, build_prop_locations: list[tuple[str, str]]) -> None:
    """test if we detect different build.prop locations correctly."""

    fs = VirtualFilesystem()
    fs.makedirs("/data")
    fs.makedirs("/system")
    fs.makedirs("/vendor")
    fs.makedirs("/product")

    for prop, prop_file in build_prop_locations:
        fs.map_file(prop, absolute_path("_data/plugins/os/unix/linux/android/{prop_file}"))
```
Maybe a bit nicer? Not sure.